schema/ref.go

```diff
@@ -92,10 +106,27 @@ func (r *RefSchema) ApplyScope(scope Scope) {
 	r.referencedObjectCache = referencedObject
 }
 
+func (r *RefSchema) ValidateReferences() error {
+	if r.referencedObjectCache != nil {
+		return nil
+	}
+
+	return BadArgumentError{
+		Message: fmt.Sprintf(
+			"Ref object reference could not find an object with ID %q in namespace %q",
+			r.IDValue,
+			r.ObjectNamespace,
+		),
+	}
+}
```

Comment:
What I find confusing is that the message complains we "could not find" ... this seems to actually mean, in this limited context, simply that we don't *have* a reference, since this method has no way to know for sure that anyone even looked.

We *have* a referenced object cache, or we don't: but a message that implies it knows *why* that might have happened seems inappropriate and potentially misleading to a mere human reading a log. 😆